mne/dipole.py

```diff
@@ -546,8 +546,16 @@ def fit_dipole(evoked, cov, bem, trans=None, min_dist=5.,
     logger.info('%d bad channels total' % len(info['bads']))
 
     # Forward model setup (setup_forward_model from setup.c)
-    megcoils, compcoils, eegels, megnames, eegnames, meg_info = \
-        _prep_channels(info, exclude='bads', accurate=accurate)
+    ch_types = [channel_type(info, idx) for idx in range(info['nchan'])]
+
+    megcoils, compcoils, megnames, meg_info = [], [], [], None
+    eegels, eegnames = [], []
+    if 'grad' in ch_types or 'mag' in ch_types:
+        megcoils, compcoils, megnames, meg_info = \
+            _prep_meg_channels(info, accurate=accurate, verbose=verbose)
+    if 'eeg' in ch_types:
+        eegels, eegnames = _prep_eeg_channels(info, exclude='bads',
+                                              verbose=verbose)
```

Comment:
What happens if neither MEG nor EEG channels were present? It used to throw a clear error, now it will not. I assume it'll produce some cryptic error later instead...